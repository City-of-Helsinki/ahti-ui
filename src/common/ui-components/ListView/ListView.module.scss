.divider {
  height: 1px;
  background: var(--hds-ui-color-black);
}

.line {
  border: 0.5px solid var(--hds-ui-color-black-10);
}
