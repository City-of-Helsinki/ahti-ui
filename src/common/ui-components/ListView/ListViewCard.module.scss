@import '../../../styles/icons';

.container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  background-color: var(--hds-ui-color-white);
  cursor: pointer;
}

.infoContainer {
  padding: 0.5rem;
  display: flex;
  flex-direction: row;
  flex: 2;
}

.infoTextContainer {
  margin-left: 1rem;
  display: flex;
  flex-direction: column;
}

.heading {
  font-size: 18px;
  line-height: 23px;
  font-weight: 700;
}

.shortDescription {
  font-size: 12px;
  line-height: 16px;
}

.image {
  object-fit: cover;
  min-height: 100%;
  width: 100px;
}
