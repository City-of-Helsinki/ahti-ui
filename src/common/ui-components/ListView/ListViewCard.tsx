import React from 'react';
import { useTranslation } from 'react-i18next';

import { Feature } from '../../../domain/api/generated/types.d';
import CategoryIcon from '../CategoryIcon/CategoryIcon';
import { getFirstImageUrl } from '../../utils/images';
import styles from './ListViewCard.module.scss';

export interface ListViewCardProps {
  readonly feature: Feature;
  onClick(): void;
}

const ListViewCard: React.FC<ListViewCardProps> = ({ feature, onClick }) => {
  const { t } = useTranslation();
  const category = feature?.properties?.category?.id;
  const name = feature?.properties?.name;
  const shortDescription = feature?.properties?.shortDescription;
  const imageUrl = getFirstImageUrl(feature?.properties?.images, null);

  return (
    <div
      className={styles.container}
      role={'button'}
      onClick={onClick}
      tabIndex={0}
      aria-label={`${t('list_view.open')}: ${name}`}
    >
      <div className={styles.infoContainer}>
        <div>
          <CategoryIcon category={category} className={styles.largeIcon} />
        </div>
        <div className={styles.infoTextContainer}>
          <h2 className={styles.heading}>{name}</h2>
          <div className={styles.shortDescription}>{shortDescription}</div>
        </div>
      </div>
      {imageUrl && (
        <img
          src={imageUrl}
          className={styles.image}
          alt={`${t('list_view.image_alt')}: ${name}`}
          width={100}
        />
      )}
    </div>
  );
};

export default ListViewCard;
