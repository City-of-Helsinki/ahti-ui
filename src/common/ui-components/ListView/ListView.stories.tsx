import React from 'react';
import ListView from './ListView';
import { useFeaturesQuery } from '../../../domain/api/generated/types.d';
import { ApolloProvider } from '@apollo/react-common';
import graphQLClient from '../../../domain/api';
import { featuresLens } from '../../utils/lenses';
import mockFeature from './__tests__/mockFeature';

export default {
  title: 'ListView',
  component: ListView,
  decorators: [
    (storyFn: () => React.ReactNode) => (
      <ApolloProvider client={graphQLClient}>{storyFn()}</ApolloProvider>
    )
  ]
};

export const MockedFeatures = () => {
  return <ListView features={Array(10).fill(mockFeature)} />;
};

MockedFeatures.story = {
  name: 'Mocked data'
};

const TwentyFeatures = () => {
  const { data } = useFeaturesQuery();
  return (
    <>{data && <ListView features={featuresLens.get(data).slice(0, 20)} />}</>
  );
};

// The component needs to be wrapped,
// otherwise the decorator doesn't get applied for whatever reason.
export const TwentyFeaturesWrapper = () => {
  return <TwentyFeatures />;
};

TwentyFeaturesWrapper.story = {
  name: '20 Features'
};
