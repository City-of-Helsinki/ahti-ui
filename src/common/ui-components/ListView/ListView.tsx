import React from 'react';
import classNames from 'classnames';

import ListViewCard from './ListViewCard';
import { Feature } from '../../../domain/api/generated/types.d';
import styles from './ListView.module.scss';

export interface ListViewProps {
  readonly className?: string;
  readonly features: Feature[];
  onClick?(feature: Feature): void;
}

const ListView: React.FC<ListViewProps> = ({
  className,
  features,
  onClick,
}) => {
  return (
    <div className={classNames(className)}>
      {features.length > 0 && <hr className={styles.line} />}
      {features.map((feature: Feature, id: number) => {
        return (
          <React.Fragment key={id}>
            <ListViewCard
              key={id}
              feature={feature}
              onClick={() => onClick && onClick(feature)}
            />
            <hr className={styles.line} />
          </React.Fragment>
        );
      })}
    </div>
  );
};

export default ListView;
