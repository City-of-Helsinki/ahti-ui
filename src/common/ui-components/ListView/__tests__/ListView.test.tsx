import React from 'react';
import { shallow } from 'enzyme';

import ListView from '../ListView';
import mockFeature from './mockFeature';

describe('ListView', () => {
  const wrapper = shallow(<ListView features={Array(10).fill(mockFeature)} />);

  it('renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
