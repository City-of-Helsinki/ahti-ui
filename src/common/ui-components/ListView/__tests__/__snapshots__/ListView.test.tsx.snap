// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView renders the component 1`] = `"<div class=\\"\\"><hr class=\\"line\\"/><div class=\\"container\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"Open item: Pihlajasaari\\"><div class=\\"infoContainer\\"><div><svg class=\\"largeIcon\\">icon-location.svg</svg></div><div class=\\"infoTextContainer\\"><h2 class=\\"heading\\">Pihlajasaari</h2><div class=\\"shortDescription\\">Lyhyt kuvaus</div></div></div><img src=\\"https://edit.myhelsinki.fi/sites/default/files/styles/api_1980x1020/public/2017-05/Juhannus_Pihlajasaaressa_dewberry__3476.jpg?h=30c08e7f&amp;itok=B9g8B9Em\\" class=\\"image\\" alt=\\"Image: Pihlajasaari\\" width=\\"100\\"/></div><hr class=\\"line\\"/><div class=\\"container\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"Open item: Pihlajasaari\\"><div class=\\"infoContainer\\"><div><svg class=\\"largeIcon\\">icon-location.svg</svg></div><div class=\\"infoTextContainer\\"><h2 class=\\"heading\\">Pihlajasaari</h2><div class=\\"shortDescription\\">Lyhyt kuvaus</div></div></div><img src=\\"https://edit.myhelsinki.fi/sites/default/files/styles/api_1980x1020/public/2017-05/Juhannus_Pihlajasaaressa_dewberry__3476.jpg?h=30c08e7f&amp;itok=B9g8B9Em\\" class=\\"image\\" alt=\\"Image: Pihlajasaari\\" width=\\"100\\"/></div><hr class=\\"line\\"/><div class=\\"container\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"Open item: Pihlajasaari\\"><div class=\\"infoContainer\\"><div><svg class=\\"largeIcon\\">icon-location.svg</svg></div><div class=\\"infoTextContainer\\"><h2 class=\\"heading\\">Pihlajasaari</h2><div class=\\"shortDescription\\">Lyhyt kuvaus</div></div></div><img src=\\"https://edit.myhelsinki.fi/sites/default/files/styles/api_1980x1020/public/2017-05/Juhannus_Pihlajasaaressa_dewberry__3476.jpg?h=30c08e7f&amp;itok=B9g8B9Em\\" class=\\"image\\" alt=\\"Image: Pihlajasaari\\" width=\\"100\\"/></div><hr class=\\"line\\"/><div class=\\"container\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"Open item: Pihlajasaari\\"><div class=\\"infoContainer\\"><div><svg class=\\"largeIcon\\">icon-location.svg</svg></div><div class=\\"infoTextContainer\\"><h2 class=\\"heading\\">Pihlajasaari</h2><div class=\\"shortDescription\\">Lyhyt kuvaus</div></div></div><img src=\\"https://edit.myhelsinki.fi/sites/default/files/styles/api_1980x1020/public/2017-05/Juhannus_Pihlajasaaressa_dewberry__3476.jpg?h=30c08e7f&amp;itok=B9g8B9Em\\" class=\\"image\\" alt=\\"Image: Pihlajasaari\\" width=\\"100\\"/></div><hr class=\\"line\\"/><div class=\\"container\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"Open item: Pihlajasaari\\"><div class=\\"infoContainer\\"><div><svg class=\\"largeIcon\\">icon-location.svg</svg></div><div class=\\"infoTextContainer\\"><h2 class=\\"heading\\">Pihlajasaari</h2><div class=\\"shortDescription\\">Lyhyt kuvaus</div></div></div><img src=\\"https://edit.myhelsinki.fi/sites/default/files/styles/api_1980x1020/public/2017-05/Juhannus_Pihlajasaaressa_dewberry__3476.jpg?h=30c08e7f&amp;itok=B9g8B9Em\\" class=\\"image\\" alt=\\"Image: Pihlajasaari\\" width=\\"100\\"/></div><hr class=\\"line\\"/><div class=\\"container\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"Open item: Pihlajasaari\\"><div class=\\"infoContainer\\"><div><svg class=\\"largeIcon\\">icon-location.svg</svg></div><div class=\\"infoTextContainer\\"><h2 class=\\"heading\\">Pihlajasaari</h2><div class=\\"shortDescription\\">Lyhyt kuvaus</div></div></div><img src=\\"https://edit.myhelsinki.fi/sites/default/files/styles/api_1980x1020/public/2017-05/Juhannus_Pihlajasaaressa_dewberry__3476.jpg?h=30c08e7f&amp;itok=B9g8B9Em\\" class=\\"image\\" alt=\\"Image: Pihlajasaari\\" width=\\"100\\"/></div><hr class=\\"line\\"/><div class=\\"container\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"Open item: Pihlajasaari\\"><div class=\\"infoContainer\\"><div><svg class=\\"largeIcon\\">icon-location.svg</svg></div><div class=\\"infoTextContainer\\"><h2 class=\\"heading\\">Pihlajasaari</h2><div class=\\"shortDescription\\">Lyhyt kuvaus</div></div></div><img src=\\"https://edit.myhelsinki.fi/sites/default/files/styles/api_1980x1020/public/2017-05/Juhannus_Pihlajasaaressa_dewberry__3476.jpg?h=30c08e7f&amp;itok=B9g8B9Em\\" class=\\"image\\" alt=\\"Image: Pihlajasaari\\" width=\\"100\\"/></div><hr class=\\"line\\"/><div class=\\"container\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"Open item: Pihlajasaari\\"><div class=\\"infoContainer\\"><div><svg class=\\"largeIcon\\">icon-location.svg</svg></div><div class=\\"infoTextContainer\\"><h2 class=\\"heading\\">Pihlajasaari</h2><div class=\\"shortDescription\\">Lyhyt kuvaus</div></div></div><img src=\\"https://edit.myhelsinki.fi/sites/default/files/styles/api_1980x1020/public/2017-05/Juhannus_Pihlajasaaressa_dewberry__3476.jpg?h=30c08e7f&amp;itok=B9g8B9Em\\" class=\\"image\\" alt=\\"Image: Pihlajasaari\\" width=\\"100\\"/></div><hr class=\\"line\\"/><div class=\\"container\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"Open item: Pihlajasaari\\"><div class=\\"infoContainer\\"><div><svg class=\\"largeIcon\\">icon-location.svg</svg></div><div class=\\"infoTextContainer\\"><h2 class=\\"heading\\">Pihlajasaari</h2><div class=\\"shortDescription\\">Lyhyt kuvaus</div></div></div><img src=\\"https://edit.myhelsinki.fi/sites/default/files/styles/api_1980x1020/public/2017-05/Juhannus_Pihlajasaaressa_dewberry__3476.jpg?h=30c08e7f&amp;itok=B9g8B9Em\\" class=\\"image\\" alt=\\"Image: Pihlajasaari\\" width=\\"100\\"/></div><hr class=\\"line\\"/><div class=\\"container\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"Open item: Pihlajasaari\\"><div class=\\"infoContainer\\"><div><svg class=\\"largeIcon\\">icon-location.svg</svg></div><div class=\\"infoTextContainer\\"><h2 class=\\"heading\\">Pihlajasaari</h2><div class=\\"shortDescription\\">Lyhyt kuvaus</div></div></div><img src=\\"https://edit.myhelsinki.fi/sites/default/files/styles/api_1980x1020/public/2017-05/Juhannus_Pihlajasaaressa_dewberry__3476.jpg?h=30c08e7f&amp;itok=B9g8B9Em\\" class=\\"image\\" alt=\\"Image: Pihlajasaari\\" width=\\"100\\"/></div><hr class=\\"line\\"/></div>"`;
